### PR TITLE
Added isChannelAdmin middleware

### PR DESCRIPTION
**Pull request by Jarviss77**
Issue: #31

### CODE REVIEW

File: src/middlewares/isChannelAdmin.middleware.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import  Channel  from '../models/channel.models.js';
+import Workspace from '../models/workspace.models.js';
+import USER_ROLE from '../enums/userRoles.enums.js';
+import { response_404, response_403, response_500 } from '../utils/responseCodes.utils.js';
+export async function isChannelAdmin(req, res, next) {
+    try {
+        const user = req.body.user;
+        const channel = await Channel.findById(req.params.id);
+        const workspace = await Workspace.findById(channel.workspace._id);
```
Comment (automatefreely) [inline src/middlewares/isChannelAdmin.middleware.js]: workspace is not populated so you can't get _id from channel.workspace

File: src/middlewares/isChannelAdmin.middleware.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+        if (!channel) {
```
Comment (automatefreely) [inline src/middlewares/isChannelAdmin.middleware.js:10]: You can put this check before getting workspace from channel.

File: src/middlewares/isChannelAdmin.middleware.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            return response_404(res, 'Channel not found');
+        }
+
+        const isAdminChannel = user.channels.some((channel) => channel._id === req.params.id && channel.role === USER_ROLE.ADMIN);
```
Comment (automatefreely) [inline src/middlewares/isChannelAdmin.middleware.js]: You should convert  channel._id to string

**Review by automatefreely — COMMENTED** [review]
Nicely written! 
Review the following problems

**Comment by Jarviss77** [comment]
pls review

File: src/middlewares/isChannelAdmin.middleware.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import  Channel  from '../models/channel.models.js';
```
Comment (automatefreely) [inline src/middlewares/isChannelAdmin.middleware.js]: Do not use import as type in package.json is not set to module. in all other files we have use require import format.

**Pull request merged**